src/connection/coniql.ts

```diff
@@ -234,16 +238,45 @@ const PV_MUTATION = gql`
   }
 `;
 
-export class ConiqlPlugin implements Connection {
-  private client: ApolloClient<NormalizedCacheObject>;
-  private onConnectionUpdate: ConnectionChangedCallback;
-  private onValueUpdate: ValueChangedCallback;
+const DEVICE_SUBSCRIPTION = gql`
+  query deviceQuery($pvDevice: ID!) {
+    getDevice(id: $pvDevice) {
+      id
+      children(flatten: true) {
+        name
+        label
+        child {
+          __typename
+          ... on Channel {
+            id
+          }
+          ... on Device {
+            id
+          }
+          ... on Group {
+            layout
+            children {
+              name
+            }
+          }
+        }
+      }
+    }
+  }
+`;
```

Comment:
This should be all you ever have to do, on connection to coniql. And it is static, so can't change without restarting coniql, so you never need to refetch it. This means no subscribing or unsubscribing, the first time you are asked about the contents of a Device you run this query, then subscribe to the Channels give.

Reply:
So yes, what @willrogers said is correct